code/go/0chain.net/smartcontract/dbs/event/edb_transaction.go

```diff
@@ -0,0 +1,15 @@
+package event
+
+import (
+	"0chain.net/smartcontract/dbs"
+	"gorm.io/gorm"
+)
+
+type edbTransaction struct {
```

Comment:
I use this struct to avoid updating all the database statements in [dbs/event](https://github.com/0chain/0chain/tree/staging/code/go/0chain.net/smartcontract/dbs/event). 

By replacing the Store in the event database, we can make the `edb.Store.Get()` calls return the transaction, not the underlying `gorm.DB`.